ghga-de/datameta
# Test/user

"""Tests user management for the API
    +: Changes that should work
    -: Changes that should not work
"""

"""Test 1: Namechange
a) user_a:
    + change name of user_a
    - change name of user_c
b) group_x_admin:
    + change name of user_a
    - change name of user_c
c) admin:
    + change name of user_a
"""

"""Test 2: Group Change
a) user_a:
    - change group of user_a
b) group_x_admin:
    - change group of group_x_admin
    - change group of user_a
    - change group of user_c to group_x
c) admin:
    + change group of admin
    + change group of user_a
"""

"""Test 3: Admin settings change
a) user_a:
    - make user_a admin
    - remove admin status of admin
b) group_x_admin:
    + make user_a group_admin
    + remove user_a as group_admin
    - make group_x_admin site admin
    - make user_a site admin
    - remove admin status of admin
c) admin:
    + make user_a group & site admin 
    + remove admin status of user_a
"""

"""Test 4: Enabled settings change:
#Note, that the enabled setting currently does not do anything. Once this changes, these tests have to be expanded.
a) user_a:
    - disable/enable user_a
b) group_x_admin:
    + disable/enable user_a
    - (admin: make user_b site admin) disable/enable user_b # This should not work. Change, when changed in the code
    - disable/enable user_c
c) admin:
    + disable/enable user_b
"""
